libcxx/include/ios

```diff
@@ -605,7 +643,7 @@ template <class _CharT, class _Traits>
 inline _LIBCPP_HIDE_FROM_ABI void basic_ios<_CharT, _Traits>::init(basic_streambuf<char_type, traits_type>* __sb) {
   ios_base::init(__sb);
   __tie_  = nullptr;
-  __fill_ = traits_type::eof();
+  __fill_ = widen(' ');
```

Comment:
There is some history to why widen is not called here:
https://github.com/llvm/llvm-project/commit/3320c058bf4e7b827d8546b31881ff043c3a44df

Reply:
It would be worth adding this as a comment.  It's so easy to think calling widen() here is the better thing to do.

Reply:
Removed `__fill_ = widen(' ');` because the original statement `__fill_ = traits_type::eof();` sets the fill value to `traits_type::eof()` to indicate it is uninitialized. This is now accomplished by constructors, `_SentinelValueFill() : __fill_val_(_Traits::eof()) { }` and ` _FillHelper() : __set_(false) { }`.

Reply:
> This is now accomplished by constructors

basic_ios has a bunch of rather interesting initialization requirements according to the standard (related to why it has an init function at all), and the constructor isn't really supposed to initialize in some cases.

I think to preserve that, and the original flow of the code, you should add an Init function to the both fill types in place of the constructor and call it here.

Reply:
Thanks for the insight! Changed to use an init function as suggested.